tests/template_tests/test_base.py

```diff
@@ -1,8 +1,47 @@
 from django.template import Context, Template, Variable, VariableDoesNotExist
+from django.template.base import DebugLexer, Lexer, TokenType
 from django.test import SimpleTestCase
 from django.utils.translation import gettext_lazy
 
 
+class LexerTestMixin:
+    template_string = ('text\n'
+                       '{% if test %}{{ varvalue }}{% endif %}'
+                       '{#comment {{not a var}} %{not a block}% #}')
+
+    expected_token_tuples = [
+        # (token_type, contents, lineno, position)
+        (TokenType.TEXT, 'text\n', 1, (0, 5)),
+        (TokenType.BLOCK, 'if test', 2, (5, 18)),
+        (TokenType.VAR, 'varvalue', 2, (18, 32)),
+        (TokenType.BLOCK, 'endif', 2, (32, 43)),
+        (TokenType.COMMENT, 'comment {{not a var}} %{not a block}%', 2, (43, 85)),
+    ]
+
+    def test_tokenize(self):
+        tokens = self.lexer_class(self.template_string).tokenize()
+        token_tuples = [(t.token_type, t.contents, t.lineno, t.position) for t in tokens]
+        self.assertEqual(token_tuples, self.make_expected())
+
+    def make_expected(self):
+        raise NotImplementedError
```

Comment:
I thought I gave this comment, but maybe I forgot to  hit "add." You might as well add a message to this exception like is done in the CSRF tests, for example: https://github.com/django/django/blob/8208381ba6a3d1613bb746617062ccf1a6a28591/tests/csrf_tests/tests.py#L198